Aztec/Classes/TextKit/ColorProvider.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import UIKit
+
+/// A class that provides colors to the formatter in TextKit.
+/// By default only the standard html colors are provided, but new color can be added by using the `add(namedColors:)` method.
+public class ColorProvider {
+
+
+    private init() {
+
+    }
+
+    public static var shared = ColorProvider()
+
+    private lazy var htmlColors: [String: UIColor] = {
+        guard let colorDataURL = Bundle.aztecBundle.url(forResource: "html_colors", withExtension: "json"),
+            let data = try? Data(contentsOf: colorDataURL),
+            let object = try? JSONSerialization.jsonObject(with: data, options: .allowFragments),
+            let colorDict = object as? [String: String] else {
+                return [String:UIColor]()
+        }
+
+        return colorDict.compactMapValues { (value) -> UIColor? in
+            return UIColor.init(hexString: value)
+        }
+    }()
+
+    private var extraColors = [String:UIColor]()
+
+    /// Adds extra named colors to be used by the `color(named:)` method.
+    ///  If the there is a name collision the value on the added dictionary will overwrite the current color.
+    /// - Parameter namedColors: a dictionary of colors keyed by name
+    public func add(namedColors: [String: UIColor]) {
+        extraColors.merge(namedColors) { (base, other) -> UIColor in
+            return other
+        }
+    }
```

Comment:
By the current implementation, if a user adds a new color called `"yellow"` with a different yellow tone (for example), when fetching the `color(named: "yellow")`, it will still return the default yellow value.

This sounds like it's fine since we are following the standard colours for the default. But it might be unexpected.

Not sure if this is really a problem. What do you think?

Reply:
I want to keep the default HTML colors "static", and allow to add extra colors, for example if Gutenberg defines some standard CSS colors that we will need to use.